iroh-relay/src/main.rs

```diff
@@ -362,46 +391,83 @@ async fn main() -> Result<()> {
+    cfg: &Config,
+) -> Result<Option<relay::TlsConfig<std::io::Error, std::io::Error>>> {
+    if cfg.tls.is_none() {
+        return Ok(None);
+    }
+    let tls = cfg.tls.as_ref().expect("checked");
+    let server_config = rustls::ServerConfig::builder_with_provider(std::sync::Arc::new(
+        rustls::crypto::ring::default_provider(),
+    ))
+    .with_safe_default_protocol_versions()
+    .expect("protocols supported by ring")
+    .with_no_client_auth();
+    let (cert_config, server_config) = match tls.cert_mode {
+        CertMode::Manual => {
+            let cert_path = tls.cert_path();
+            let key_path = tls.key_path();
+            // Could probably just do this blocking, we're only starting up.
+            let (private_key, certs) = tokio::task::spawn_blocking(move || {
+                let key = load_secret_key(key_path)?;
+                let certs = load_certs(cert_path)?;
+                anyhow::Ok((key, certs))
             })
+            .await??;
+            let server_config = server_config.with_single_cert(certs.clone(), private_key)?;
+            (relay::CertConfig::Manual { certs }, server_config)
+        }
+        CertMode::LetsEncrypt => {
+            let hostname = tls
+                .hostname
+                .clone()
+                .context("LetsEncrypt needs a hostname")?;
+            let contact = tls
+                .contact
+                .clone()
+                .context("LetsEncrypt needs a contact email")?;
+            let config = AcmeConfig::new(vec![hostname.clone()])
+                .contact([format!("mailto:{}", contact)])
+                .cache_option(Some(DirCache::new(tls.cert_dir())))
+                .directory_lets_encrypt(tls.prod_tls);
+            let state = config.state();
+            let resolver = state.resolver().clone();
+            let server_config = server_config.with_cert_resolver(resolver);
+            (relay::CertConfig::LetsEncrypt { state }, server_config)
         }
-        None => None,
     };
+    Ok(Some(relay::TlsConfig {
+        https_bind_addr: tls.https_bind_addr(cfg),
+        cert: cert_config,
+        server_config,
+    }))
+}
+
+/// Convert the TOML-loaded config to the [`relay::RelayConfig`] format.
+async fn build_relay_config(cfg: Config) -> Result<relay::ServerConfig<std::io::Error>> {
+    let relay_tls = maybe_load_tls(&cfg).await?;
+
+    let mut quic_config = None;
+    if cfg.enable_quic_addr_discovery {
+        if let Some(ref tls) = relay_tls {
+            quic_config = Some(QuicConfig::new(
+                tls.server_config.clone(),
+                tls.https_bind_addr.ip(),
+                cfg.quic_bind_port,
+            )?);
+        } else if cfg.enable_quic_local_cert {
+            let (_, server_config) = iroh_relay::server::self_signed_tls_certs_and_config();
+            quic_config = Some(QuicConfig::new(
+                server_config,
+                Ipv6Addr::UNSPECIFIED.into(),
```

Comment:
why ipv6 by default?

Reply:
quinn specific: if you bind to `Ipv6Addr::UNSPECIFIED` it will attempt to bind to both `Ipv6Addr::UNSPECIFIED` and `Ipv4Addr::UNSPECIFIED` if you can do dual-stack sockets on that machine.

Reply:
I see, can you add a comment about this?